packages/atomic/playwright-utils/cache.ts

```diff
@@ -0,0 +1,40 @@
+import {APIResponse, Request, Route} from '@playwright/test';
+import {createHash} from 'crypto';
+import {existsSync, mkdirSync, readFileSync, writeFileSync} from 'fs';
+import {join} from 'path';
+
+export const cacheFilePath = './playwright/.cache/';
+
+export const hashRequest = (url: string, body: string | null) => {
+  const str = url + body;
+  return createHash('sha512').update(str).digest('hex');
+};
+
+export const cache = async (key: string, response: APIResponse) => {
+  mkdirSync(cacheFilePath, {recursive: true});
+  const body = await response.text();
+  // TODO: check if the file already exists and throw
+  writeFileSync(join(cacheFilePath, key), body);
+  return body;
+};
+
+export const cacheHandler = async (route: Route, request: Request) => {
+  console.log('intercepted ---', request.url());
+  const {clientId, ...rest} = request.postDataJSON();
+  const key = hashRequest(request.url(), JSON.stringify(rest));
```

Comment:
**question**: Are you using the same authentication for each requests? Two exact request with the same payload may have different response depending on the authentication used.